x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/deserializer_and_serializer.test.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { cloneDeep, set } from 'lodash';
+import { SerializedPolicy } from '../../../../../common/types';
+import { deserializer } from './deserializer';
+import { createSerializer } from './serializer';
+
+describe('deserializer and serializer', () => {
+  it('preserves unknown policy settings', () => {
+    const originalPolicy: SerializedPolicy = {
+      name: 'test',
+      phases: {
+        hot: {
+          actions: {
+            rollover: {
+              max_age: '1d',
+              max_size: '10gb',
+              max_docs: 1000,
+            },
+            forcemerge: {
+              index_codec: 'best_compression',
+              max_num_segments: 22,
+            },
+          },
+          min_age: '12ms',
+        },
+        warm: {
+          actions: {
+            shrink: { number_of_shards: 12 },
+            allocate: {
+              number_of_replicas: 3,
+            },
+            set_priority: {
+              priority: 10,
+            },
+          },
+        },
+        cold: {
+          actions: {
+            allocate: { number_of_replicas: 12 },
+            freeze: {},
+            set_priority: {
+              priority: 12,
+            },
+          },
+        },
+        delete: {
+          actions: {
+            delete: {
+              delete_searchable_snapshot: true,
+            },
+            wait_for_snapshot: {
+              policy: 'test',
+            },
+          },
+        },
+      },
+    };
+
+    /**
+     * Next we insert unknown values at various places in the policy object
+     * where we know they may occur.
+     */
+    [
+      '',
+      'phases.hot.actions',
+      'phases.hot.actions.rollover',
```

Comment:
Yeah, I started writing this up and something didn't feel quite right because we would need to update this list in order for the check it is performing to remain thorough and reliable. I opted for creating a small bit of functionality that populates the entire policy with unknown values at all levels (except for one, but I left a comment about that). Let me know what you think!